pkg/hs/urbit-king/lib/Urbit/Vere/Term/Logic.hs

```diff
@@ -83,15 +83,17 @@ step st@St{..} = \case
     EvLine t -> st & recordText t
     EvSlog s -> st & recordSlog s
     EvSpin s -> st { sSpinner = s }
-    EvMove w -> st { sCurPos = min w (word $ length sLine) }
-    EvEdit t -> st { sLine = t, sCurPos = word (length t) }
-    EvMore   -> st { sLine = "", sCurPos = 0 } & recordText (sLine <> "\n")
+    EvMove p -> st { sCurPos = p }
+    EvEdit t -> if fst sCurPos == 0 then st { sLine = t }
+                else st
```

Comment:
```suggestion
    EvEdit t | (0, _) <- sCurPos -> st { sLine = t }
             | otherwise         -> st
```
and likewise below. This is called a "pattern guard." In general the existence of "guards" means you shouldn't ever need to put an if expression to the right hand side of a pattern match